src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -268,28 +316,43 @@ public void IsCompatibilitySwitchSet()
         [Fact]
         public void IsDefaultAppDomain()
         {
-            Assert.True(AppDomain.CurrentDomain.IsDefaultAppDomain());
+            RemoteInvoke(() =>
+            {
+                Assert.True(AppDomain.CurrentDomain.IsDefaultAppDomain());
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void IsFinalizingForUnload()
         {
-            Assert.False(AppDomain.CurrentDomain.IsFinalizingForUnload());
+            RemoteInvoke(() => {
+                Assert.False(AppDomain.CurrentDomain.IsFinalizingForUnload());
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void toString()
         {
             string actual = AppDomain.CurrentDomain.ToString();
+
+            // NetFx has additional line endings
+            if (PlatformDetection.IsFullFramework)
+                actual = actual.Trim();
+
             string expected = "Name:" + AppDomain.CurrentDomain.FriendlyName + Environment.NewLine + "There are no context policies.";
             Assert.Equal(expected, actual);
         }
 
         [Fact]
         public void Unload()
         {
-            AssertExtensions.Throws<ArgumentNullException>("domain", () => { AppDomain.Unload(null);});
-            Assert.Throws<CannotUnloadAppDomainException>(() => { AppDomain.Unload(AppDomain.CurrentDomain); });
+            RemoteInvoke(() => {
```

Comment:
It's trying to unload current appdomain. If this works on any platform because of the bug then lots of tests will fail, if you use RemoteInvoke only related will